Dockerfile

```diff
@@ -1,38 +1,47 @@
-FROM registry.access.redhat.com/ubi8/ubi:latest AS build
-
+FROM registry.access.redhat.com/ubi8/ubi:8.7 AS build
 ENV TNF_DIR=/usr/tnf
 ENV TNF_SRC_DIR=${TNF_DIR}/tnf-src
 ENV TNF_BIN_DIR=${TNF_DIR}/cnf-certification-test
-
 RUN mkdir ${TNF_DIR}
-
 ENV TEMP_DIR=/tmp
 
 # Install dependencies
-RUN yum install -y gcc git jq make wget
+RUN \
+	yum install --assumeyes \
+		gcc-8.5.0 \
+		git-2.31.1 \
+		jq-1.6 \
+		cmake-3.20.2 \
+		wget-1.19.5 \
+	&& yum clean all
```

Comment:
I'm torn on whether or not we need to pin these versions.  I like that we get the latest versions whenever we build our image.

Reply:
`hadolint` doesn't like `latest`, I don't thing the versions are going to be changed in `ubi:8.7`. What is the profit to work with latest? By the way the latest is ubi9.

Reply:
Is there a way to turn off the version checking in hadolint?

Reply:
There is a way, sure.

Reply:
Let's do that for now, and decide later if we want to turn on the version checking.

Reply:
In theory every image build is different today because it uses the latest packages. What if we need to reproduce exactly the same run we did a year ago with a certain version of curl, for example. For that we have release versions for go modules, for that packages and images should have versions too. And then we lock all software versions (go modules, dependent libraries, images and packages) together under the TNF version, we test and certify it. Under this lock we can always be sure we run the same software for the release.

Reply:
Yup makes sense.  We should probably have some sort of automation that runs weekly(?) for updating these image level dependencies.  I wonder if dependabot has some sort of option to scan Dockerfiles.

Reply:
https://dev.to/oracle2025/how-to-keep-a-dockerfile-updated-with-dependabot-1mdn